src/modules/memory_usage.rs

```diff
@@ -69,20 +51,19 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         )
     };
 
-    let show_swap = module
-        .config_value_bool("show_swap")
-        .unwrap_or(total_swap_kib != 0);
+    module.create_segment("symbol", &config.symbol);
 
-    module.new_segment("symbol", RAM_CHAR);
-
-    module.set_style(module_style);
-    if show_swap {
-        module.new_segment(
+    module.set_style(config.style);
+    if config.show_swap {
+        module.create_segment(
             "memory_usage",
-            &format!("{} | {}", display_mem, display_swap),
+            &SegmentConfig::new("").with_value(&format!("{} | {}", display_mem, display_swap)),
         );
     } else {
-        module.new_segment("memory_usage", &display_mem);
+        module.create_segment(
+            "memory_usage",
+            &SegmentConfig::new("").with_value(&display_mem),
```

Comment:
Could this be `&SegmentConfig::new(&display_mem)`?

Reply:
That makes a lot more sense :laughing:. I'll give it a try, thanks!